test/simple/test-child-process-stdout-flush-exit.js

```diff
@@ -52,14 +52,17 @@ if (process.argv[2] === 'child') {
 
   // check if we receive both 'hello' at start and 'goodbye' at end
   child.stdout.setEncoding('utf8');
+  var messageSequence = 0;
   child.stdout.on('data', function (data) {
-    if (data.slice(0, 6) == 'hello\n') {
+    if ((0 == messageSequence ) && (data.slice(0, 6) == 'hello\n')) {
```

Comment:
We encourage the rvalues on the left because if you miss an = you end up with an error instead of assigning when you did not mean to.  However, if that's not commonly done in the Node codebase I guess we should be consistent